subsys/bluetooth/mesh/scene_srv.c

```diff
@@ -43,6 +45,58 @@ static inline void update_page_count(struct bt_mesh_scene_srv *srv, bool vnd,
 	}
 }
 
+static const struct bt_mesh_scene_entry *
+entry_find(const struct bt_mesh_model *mod, bool vnd)
+{
+	const struct bt_mesh_scene_entry *it;
+
+	if (vnd) {
+		extern const struct bt_mesh_scene_entry
+			_bt_mesh_scene_entry_vnd_list_start[];
+		extern const struct bt_mesh_scene_entry
+			_bt_mesh_scene_entry_vnd_list_end[];
+
+		for (it = _bt_mesh_scene_entry_vnd_list_start;
```

Comment:
Why not `Z_STRUCT_SECTION_FOREACH`?

Reply:
It only works if you have a single list of entries of this struct that you want to iterate over, and does not provide any way to split the entries into two groups, like we need here